test/TestMoment.cc

```diff
@@ -0,0 +1,204 @@
+                EXPECT_FLOAT_EQ(data1[i], data2[i]);
+            }
+        }
+    }
+
+    static void GenerateMoments(const std::shared_ptr<casacore::ImageInterface<float>>& image, int moments_axis) {
+        // create casa/carta moments generators
+        casacore::LogOrigin casa_log("casa::ImageMoment", "createMoments", WHERE);
+        casacore::LogIO casa_os(casa_log);
+        casacore::LogOrigin carta_log("carta::ImageMoment", "createMoments", WHERE);
+        casacore::LogIO carta_os(carta_log);
+        casa::ImageMoments<float> casa_image_moments(*image, casa_os, true);
+        carta::ImageMoments<float> carta_image_moments(*image, carta_os, nullptr, true);
+
+        // set moment types
+        casacore::Vector<casacore::Int> moments(12);
+        moments[0] = 0;   // AVERAGE
+        moments[1] = 1;   // INTEGRATED
+        moments[2] = 2;   // WEIGHTED_MEAN_COORDINATE
+        moments[3] = 3;   // WEIGHTED_DISPERSION_COORDINATE
+        moments[4] = 4;   // MEDIAN
+        moments[5] = 6;   // STANDARD_DEVIATION
+        moments[6] = 7;   // RMS
+        moments[7] = 8;   // ABS_MEAN_DEVIATION
+        moments[8] = 9;   // MAXIMUM
+        moments[9] = 10;  // MAXIMUM_COORDINATE
+        moments[10] = 11; // MINIMUM
+        moments[11] = 12; // MINIMUM_COORDINATE
+
+        // the other settings
+        casacore::Vector<float> include_pix;
+        casacore::Vector<float> exclude_pix;
+        casacore::Bool do_temp(true);
+        casacore::Bool remove_axis(false);
+
+        // calculate moments with casa moment generator
+        casa_image_moments.setMoments(moments);
+        casa_image_moments.setMomentAxis(moments_axis);
+        casa_image_moments.setInExCludeRange(include_pix, exclude_pix);
+        auto casa_results = casa_image_moments.createMoments(do_temp, "casa_image_moments", remove_axis);
+
+        // calculate moments with carta moment generator
+        carta_image_moments.setMoments(moments);
+        carta_image_moments.setMomentAxis(moments_axis);
+        carta_image_moments.setInExCludeRange(include_pix, exclude_pix);
+        auto carta_results = carta_image_moments.createMoments(do_temp, "carta_image_moments", remove_axis);
+
+        // check the consistency of casa/carta results
+        EXPECT_EQ(casa_results.size(), carta_results.size());
+        EXPECT_EQ(carta_results.size(), moments.size());
+
+        for (int i = 0; i < casa_results.size(); ++i) {
+            auto casa_moment_image = dynamic_pointer_cast<casacore::ImageInterface<casacore::Float>>(casa_results[i]);
+            auto carta_moment_image = dynamic_pointer_cast<casacore::ImageInterface<casacore::Float>>(carta_results[i]);
+
+            EXPECT_EQ(casa_moment_image->shape().size(), carta_moment_image->shape().size());
+            CompareImageData(casa_moment_image, carta_moment_image);
+        }
+    }
+};
+
+TEST_F(MomentTest, CheckConsistency) {
+    string file_name = "data/images/fits/M17_SWex_unittest.fits";
```

Comment:
perhaps we could use our existing 10x10x10 noise image, instead of adding this file?

Reply:
That is with NaN pixel

Reply:
ok. Then I suggest we add a similar image with minimal headers / trimmed history etc. That M17 image has a huge number of headers that are not needed

Reply:
I pushed an updated FITS file without those lengthy history in header. Please check.

Reply:
Can you do the same thing with the other image as well?

Reply:
done with 7318425